### DISCUSSION TITLE
Allow strings (const char*, char[128) in unmanaged structs without the struct becoming managed

### DISCUSSION THREAD

**Discussion by Rosentti** [dotnet/csharplang#7420] [Language Ideas]
I have a struct that self-references itself in a linked list fashion:
```c#
[StructLayout(LayoutKind.Sequential)]
public unsafe struct ConCommandBase {
    public void *vtable;
    public ConCommandBase *pNext;
    public bool bRegistered;
    public string pszName;
    public string pszHelpString;
    public int nFlags;
    public ConCommandBase *pConCommandBases;
    public IConCommandBaseAccessor *pAccessor; 
};
```

However this causes warnings about `This takes the address of, gets the size of, or declares a pointer to a managed type ('ConCommandBase')`

Here's the C++ class:
```c++
class ConCommandBase {
public:
  void *vtable;
  ConCommandBase *pNext;
  bool bRegistered;
  const char *pszName;
  const char *pszHelpString;
  int nFlags;
  ConCommandBase *pConCommandBases;
  IConCommandBaseAccessor *pAccessor; 
};
```

It would be really nice to be able to keep the strings there, easily available instead of just replacing them with IntPtr's. 
`[MarshalAs(UnmanagedType.LPUTF8Str)]` doesn't help either.

**Comment by yaakov-h** [comment]
Huh, I never expected to see Source Engine code in csharplang 😄 

I would have thought this would be better as a class rather than a struct.

**Comment by Rosentti** [comment]
I've actually decided against using that struct's fields at all, as I'm using JIT to generate a class from the class pointer, which also automatically marshals strings and stuff in methods, which did contain GetName and GetHelpString. Also, it's even more outrageous than source engine stuff, it's an open source GUI for the Steam client.
